src/components/molecules/UiInteractiveSvg/_internal/UiInteractiveSvgElement.vue

```diff
@@ -22,42 +22,47 @@ import {
   inject,
   provide,
   ref,
+  isRef,
 } from 'vue';
 import type { Ref } from 'vue';
-import type { AttrsFunc } from '../UiInteractiveSvg.vue';
+import type { InteractiveSvgFunc } from '../UiInteractiveSvg.vue';
+import type { DefineAttrsProps } from '../../../../types';
 
-defineProps({
+export interface InteractiveSvgElementProps {
   /**
-   *  Use this props to set component tag.
-   */
-  tag: {
-    type: String,
-    default: 'path',
-  },
-});
-const setAttrs: AttrsFunc | Ref<AttrsFunc> | undefined = inject('elementsAttrs');
-const elementsAttrs = (attrs: Record<string, unknown>): Record<string, unknown> => {
-  const elementsAttrs = (setAttrs as Ref<AttrsFunc>).value || setAttrs;
+  *  Use this props to set component tag.
+  */
+  tag?: string;
+}
+export type InteractiveSvgElementAttrsProps = DefineAttrsProps<InteractiveSvgElementProps>;
+
+withDefaults(defineProps<InteractiveSvgElementProps>(), { tag: 'path' });
+const setAttrs = inject<InteractiveSvgFunc | Ref<InteractiveSvgFunc>>('elementsAttrs', () => ({}));
+const elementsAttrs: InteractiveSvgFunc = (attrs) => {
+  if (!setAttrs) {
+    return {};
+  }
+  const elementsAttrs = isRef<InteractiveSvgFunc>(setAttrs) ? setAttrs.value : setAttrs;
   const {
     elementsAttrs: elementsAttrsForNested, ...rest
   } = elementsAttrs(attrs);
   if (elementsAttrsForNested) {
-    nestedElementsAttrs.value = elementsAttrsForNested as AttrsFunc;
+    nestedElementsAttrs.value = elementsAttrsForNested;
   } else {
-    nestedElementsAttrs.value = elementsAttrs as AttrsFunc;
+    nestedElementsAttrs.value = elementsAttrs;
   }
   return {
     ...attrs,
     ...rest,
   };
 };
-const nestedElementsAttrs = ref<AttrsFunc>(() => ({}));
+const nestedElementsAttrs = ref<InteractiveSvgFunc>(() => ({}));
 const parentComponent = getCurrentInstance()?.parent;
 if (!parentComponent || (parentComponent.type.name !== 'UiInteractiveSvg' && parentComponent.type.name !== 'UiInteractiveSvgElement')) {
   if (process.env.NODE_ENV !== 'production') {
     throw new Error('UiInteractiveSvgElement has to be child of UiInteractiveSvg or nested in another UiInteractiveSvgElement');
   }
 }
-provide('elementsAttrs', nestedElementsAttrs);
+provide<Ref<InteractiveSvgFunc> | InteractiveSvgFunc>('elementsAttrs', nestedElementsAttrs);
```

Comment:
Q: Why is this either `ref` or _not-ref_? 🤔

Reply:
In the case of nested elements in elements, we provide 'computed', which is why 'ref' is needed.  I will try to simplify it. 😉